pkg/server/http/utils.go

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package http
+
+import "net/http"
+
+// ResponseWriter wraps around http.ResponseWriter to capture the status code of the response.
+type ResponseWriter struct {
```

Comment:
Maybe we need a more descriptive name? ObservableResponseWriter? ResponseWriterWithStatus?

Reply:
Let me replace it with `ResponseWriterWithStatus`